### PR TITLE
EMR Notebook operator

### LINKED ISSUE #14962: Create EMR Notebook execution operators
**Description**

Similar to [EMR Jobflow operator](https://github.com/apache/airflow/blob/4e8f9cc8d02b29c325b8a5a76b4837671bdf5f68/airflow/providers/amazon/aws/operators/emr_create_job_flow.py) and [EMR Jobflow sensor](https://github.com/apache/airflow/blob/4e8f9cc8d02b29c325b8a5a76b4837671bdf5f68/airflow/providers/amazon/aws/sensors/emr_job_flow.py), we should add operator and sensor that interact with [EMR Notebook execution](https://docs.amazonaws.cn/en_us/emr/latest/ManagementGuide/emr-managed-notebooks-headless-cli.html) and abstracts all amazon logic from the airflow user.

**Use case / motivation**
This will allow airflow users use operator and sensor built for EMR Notebook execution.

**Are you willing to submit a PR?**
Yes. We already have some [code](https://github.com/aws-samples/emr-studio-samples/blob/main/airflow_dag_for_execution/simple_dag.py#L32-L75) to start with.

**Related Issues**
Not tightly related, but we would like to follow the same standard/pattern with the [feature request for EKS operator](https://github.com/apache/airflow/issues/8544)

### PR DESCRIPTION
**Pull request by syedahsn**
This PR introduces 2 new EMR Operators `EmrStartNotebookExecutionOperator` and `EmrStopNotebookExecutionOperator` to control the execution of notebooks in an EMR cluster. The PR also includes a sensor to check the state of the notebook execution, as well as unit tests for all new operators and sensors. The PR also includes a system test to show the functionality of the operators and sensors.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/providers/amazon/aws/sensors/test_emr_notebook_execution.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        mock_conn.describe_notebook_execution.return_value = self._generate_response("FAILED", error_reason)
+        sensor = EmrNotebookExecutionSensor(
+            task_id="test_task",
+            poke_interval=0,
+            notebook_execution_id="test-execution-id",
+        )
+        with pytest.raises(AirflowException) as ex_message:
+            sensor.poke(None)
+        mock_conn.describe_notebook_execution.assert_called_once_with(NotebookExecutionId="test-execution-id")
+        assert str(ex_message.value) == "EMR job failed " + error_reason
```
Comment (Taragolis) [inline tests/providers/amazon/aws/sensors/test_emr_notebook_execution.py]: You could validate exception message within [pytest.raises](https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions)
```suggestion
        with pytest.raises(AirflowException, match=fr"EMR job failed: {error_reason}"):
            sensor.poke(None)
        mock_conn.describe_notebook_execution.assert_called_once_with(NotebookExecutionId="test-execution-id")
```

File: tests/providers/amazon/aws/operators/test_emr_notebook.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+            notebook_execution_name=PARAMS["NotebookExecutionName"],
+            notebook_params=PARAMS["NotebookParams"],
+            notebook_instance_security_group_id=PARAMS["NotebookInstanceSecurityGroupId"],
+            master_instance_security_group_id=PARAMS["ExecutionEngine"]["MasterInstanceSecurityGroupId"],
+            tags=PARAMS["Tags"],
+            wait_for_completion=True,
+        )
+        with pytest.raises(AirflowException) as ex_message:
+            op.execute(None)
+        assert "Notebook Execution reached failure state FAILED." in str(ex_message.value)
```
Comment (Taragolis) [inline tests/providers/amazon/aws/operators/test_emr_notebook.py]:
```suggestion
        with pytest.raises(AirflowException, match=r"Notebook execution reached failure state FAILED\."):
            op.execute(None)
```

File: tests/providers/amazon/aws/sensors/test_emr_notebook_execution.py (hunk 1 above)
Comment (syedahsn) [reply tests/providers/amazon/aws/sensors/test_emr_notebook_execution.py]: Nice, I didn't know I could do that. Thanks!

**Comment by syedahsn** [comment]
@Taragolis Can you take a look at this PR again when you have time? Thanks!

**Review by Taragolis — APPROVED** [review]
👍

**Review by potiuk — APPROVED** [review]

**Pull request merged**